### PR TITLE
Added TIOCSWINSZ to struct_info and added corner case stub

### PR DESCRIPTION
**Pull request by Aearnus**
This fixes issue https://github.com/kripken/emscripten/issues/6463 by adding the incorrectly missing `TIOCSWINSZ` (which is defined in `system/lib/libc/musl/arch/emscripten/bits/ioctl.h`) to the `struct_info.json` file and handling it in the `ioctl()` syscall.

### CODE REVIEW

**Comment by kripken** [comment]
Thanks! Looks good. Please just add yourself to AUTHORS, and add a test (best thing is to grep in `tests/` for similar things, like another ioctl call very much like that one, and add a line in there to test this, and confirm it fails before your fix).

**Comment by Aearnus** [comment]
Done & tests passed.

File: tests/test_other.py [hunk 1]
```diff
@@ -8229,3 +8229,9 @@ def test_autotools_shared_check(self):
       except OSError:
         # Ignore missing python aliases.
         pass
+        
+  def test_ioctl_window_size(self):
+      run_process([PYTHON, EMCC, path_from_root('tests', 'ioctl', 'ioctl_window_size.c')])
```
Comment (kripken) [inline tests/test_other.py]: these 4 lines can be replaced with
```
    self.do_other_test(os.path.join('other', 'ioctl'))
```
, but you'll need to rename the `c` file to `test.cpp` and the output file to `test.out`.

Comment (kripken) [reply tests/test_other.py]: Also, would be better to do `self.do_other_test(os.path.join('other', 'ioctl', 'window_size'))` (add a subdir `window_size`).

Comment (Aearnus) [reply tests/test_other.py]: Updated it. My local copy didn't have `self.do_other_test` so I pulled it from upstream -- you should be able to remove any changes this made while merging.

**Comment by kripken** [comment]
Looks like you need to merge upstream incoming again, as there are conflicts.

**Comment by Aearnus** [comment]
@kripken Merged upstream.

**Comment by kripken** [comment]
Thanks, perfect!

**Pull request merged**